components/ai_chat/resources/page/constants/file_types.ts

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+// Supported image file types for AI Chat uploads (matches C++ backend)
+export const SUPPORTED_IMAGE_TYPES = [
+  'image/png',
+  'image/jpeg',
+  'image/jpg',
+  'image/webp'
+] as const
```

Comment:
I think the `as const` here actually makes things a bit more complicated - I think `string[]` is probably a better type as we don't need to do any type algebra with it :smile: Then we'll be able to remove the `as typeof ....` cast below

Reply:
Fixed in 870bfc15d7d8a5ba13d7ae76d181d74fcabfd6a2